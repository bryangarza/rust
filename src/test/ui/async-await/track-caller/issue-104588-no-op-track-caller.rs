// check-pass
// edition:2021

#[track_caller] //~ WARN `#[track_caller]` on async functions is a no-op, unless the `closure_track_caller` feature is enabled
async fn foo() {}

fn main() {
    foo();
}
