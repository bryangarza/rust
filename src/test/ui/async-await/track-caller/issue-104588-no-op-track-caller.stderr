warning: `#[track_caller]` on async functions is a no-op, unless the `closure_track_caller` feature is enabled
  --> $DIR/issue-104588-no-op-track-caller.rs:4:16
   |
LL |   #[track_caller]
   |  ________________^
LL | | async fn foo() {}
   | |_
   |
   = note: `#[warn(ungated_async_fn_track_caller)]` on by default

warning: 1 warning emitted

